web-api/src/personalization/routes.rs

```diff
@@ -414,6 +420,59 @@ pub(crate) async fn personalize_documents_by(
     Ok(Some(all_documents))
 }
 
+#[derive(Deserialize)]
+struct SemanticSearchQuery {
+    count: Option<usize>,
+}
+
+impl SemanticSearchQuery {
+    fn document_count(&self, config: &SemanticSearchConfig) -> Result<usize, Error> {
+        let count = self.count.map_or(config.default_number_documents, |count| {
+            count.min(config.max_number_documents)
+        });
+
+        if count > 0 {
+            Ok(count)
```

Comment:
❓ partly related to this since we use the same logic for the personalized documents `count`: why do we only error for `count < 1` but not for `count > max_number_documents`? shouldn't we either error on both or none of them? i'd prefer to not error at all and just clamp the `count`, clamping should be expectable regarding the spec.

as a side note, the spec states a fixed `100` as `max_number_documents`, but it is actually dependend on the config, so i guess we should also update the spec to reflect that. this also affects the max length in the `PersonalizedDocumentsResponse` & `SemanticSearchResponse`.

Reply:
I'm not sure why we choose to go with that :thinking: 

I think the idea was something on the line of "we might always return less documents then requested and with config changes requesting more then max can always happen so no reason to error there but requesting 0 is always a mistake so lets error".

I would argue we should just return 0 documents if the user explicitly requests 0 documents :thinking:

Reply:
Also I would make this a  :pushpin:  fixing it for both cases in a follow up PR.